home/views.py

```diff
@@ -37,9 +37,15 @@ def get(self, request):
         menu = Category.objects.all()
         products = Product.objects.all().order_by('name')
         if request.GET.get('food_name'):
-            productssearch =  filter(lambda x: unidecode((request.GET.get('food_name'))).lower()  in unidecode((_(x.name))).lower()  , products)
-            return render(request, 'catalog/menu.html', {'menu': menu,'products' : productssearch})
-        return render(request, 'catalog/menu.html', {'menu': menu,'products' : products})
+            products =  filter(lambda x: unidecode((request.GET.get('food_name'))).lower()  in unidecode((_(x.name))).lower()  , products)
+            # return render(request, 'catalog/menu.html', {'menu': menu,'products' : productssearch})
+        products_array =[]
+        for product in products:
+            promotion = Promotion.objects.filter(product=product, start_date__lt=datetime.datetime.now().isoformat(), end_date__gte=datetime.datetime.now().isoformat())
+            if(len(promotion)>0):
+                products_array.append({'product':product,'promotion':(100-promotion[0].dis_percent)*product.base_price/100,'dis_percent': promotion[0].dis_percent})
```

Comment:
Sau các toán tử `:`, `,` ,... e thêm dấu cách vào nhé
Fix tương tự bên dưới

```suggestion
                products_array.append({'product': product, 'promotion': (100-promotion[0].dis_percent) * product.base_price/100, 'dis_percent': promotion[0].dis_percent})
```